lfsapi/client.go

```diff
@@ -23,20 +23,37 @@ import (
 const MediaType = "application/vnd.git-lfs+json; charset=utf-8"
 
 var (
-	UserAgent = "git-lfs"
-	httpRE    = regexp.MustCompile(`\Ahttps?://`)
+	UserAgent                 = "git-lfs"
+	httpRE                    = regexp.MustCompile(`\Ahttps?://`)
+	maxSshAuthenticateRetries = 5
 )
 
-func (c *Client) NewRequest(method string, e Endpoint, suffix string, body interface{}) (*http.Request, error) {
-	sshRes, err := c.SSH.Resolve(e, method)
-	if err != nil {
+func (c *Client) sshResolveWithRetries(e Endpoint, method string) (sshAuthResponse, error) {
```

Comment:
What do you think about changing the return type of this method to be:

```go
(*sshAuthResponse, error)
```

instead of

```go
(sshAuthResponse, error)
```

that way we can return `nil` to indicate a failure or missing `sshAuthResponse`, instead of `sshAuthResponse{}`, which could be misleading or confusing to callers of this method.